src/components/WalletModal/index.tsx

```diff
@@ -218,48 +216,41 @@ export default function WalletModal({
         return null
       }
 
-      // overwrite injected when needed
-      if (option.connector === injected) {
+      // FIXME(maybe): We publish the captured injected providers in window.ethereum.providers array and other wallets also do the same
```

Comment:
I would leave it right now as it is for now because it sounds like a lot can go wrong